src/visitor.rs

```diff
@@ -640,10 +659,7 @@ impl<'b, 'a: 'b> FmtVisitor<'a> {
                         ErrorKind::DeprecatedAttr,
                     )],
                 );
-            } else if attr.path.segments[0].ident.to_string() == "rustfmt"
-                && (attr.path.segments.len() == 1
-                    || attr.path.segments[1].ident.to_string() != "skip")
-            {
+            } else if self.is_rustfmt_skip_attr(&attr.path.segments) {
```

Comment:
Sorry @rchaser53 now I better understand this method.

It checks if the attribute is a known rustfmt attribute, and err with `BadAttr` if not.
I'm not sure that is the best place to put such logic, but a more appropriate method would be `is_known_rustfmt_attr`:

```suggestion
            } else if self.is_known_rustfmt_attr(&attr.path.segments) {
```

In addition, can you update the documentation about the `BadAttr` variant:

```
src/lib.rs
96-    DeprecatedAttr,
97-    /// Used a rustfmt:: attribute other than skip.
98-    #[fail(display = "invalid attribute")]
99:    BadAttr,
```